AU Cairo not converting cleanly

Error:

```
2019-11-04T18:56:15+00:00 ERROR Unexpected error on record <record #1 (data/auc/p15795coll13/data/15.xml #1), output_id:oai:cdm15795.contentdm.oclc.org:p15795coll13/88>
    while executing (to_field "cho_date_range_norm" at traject_configs/auc_config.rb:36)
    Record: <?xml version="1.0"?>
<record xmlns="http://www.openarchives.org/OAI/2.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <header>
    <identifier>oai:cdm15795.contentdm.oclc.org:p15795coll13/88</identifier>
    <datestamp>2016-06-12</datestamp>
    <setSpec>p15795coll13</setSpec>
  </header>
  <metadata>
    <oai_dc:dc xmlns:oai_dc="http://www.openarchives.org/OAI/2.0/oai_dc/" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.openarchives.org/OAI/2.0/oai_dc/ http://www.openarchives.org/OAI/2.0/oai_dc.xsd">
      <dc:title>French Institute of Oriental Archaeology </dc:title>
      <dc:creator>Fathy, Hassan</dc:creator>
      <dc:subject>Architecture--Designs and plans</dc:subject>
      <dc:coverage>Luxor, Egypt</dc:coverage>
      <dc:date>ca. 1970s</dc:date>
      <dc:date>1970; 1971; 1972; 1973; 1974; 1975; 1976; 1977; 1978; 1979</dc:date>
      <dc:format>architectural drawings (visual works)</dc:format>
      <dc:identifier>HF 7_.09</dc:identifier>
      <dc:title>Hassan Fathy Architectural Archives</dc:title>
      <dc:source>Rare Books and Special Collections Library, The American University in Cairo</dc:source>
      <dc:rights>Copyright 2015, American University in Cairo. All rights reserved.</dc:rights>
      <dc:rights>Access to high-resolution master file restricted to RBSCL staff only. Please contact the RBSCL at rbscl-ref@aucegypt.edu or 2615-3676 to request access to master files</dc:rights>
      <dc:identifier>http://cdm15795.contentdm.oclc.org/cdm/ref/collection/p15795coll13/id/88</dc:identifier>
    </oai_dc:dc>
  </metadata>
</record>
    Exception: NoMethodError: undefined method `uniq!' for nil:NilClass
    /opt/traject/lib/macros/date_parsing.rb:259:in `block in auc_date_range'
[ERROR] undefined method `uniq!' for nil:NilClass
/opt/traject/lib/macros/date_parsing.rb:259:in `block in auc_date_range': undefined method `uniq!' for nil:NilClass (NoMethodError)
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer/step.rb:140:in `block in execute'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer/step.rb:135:in `each'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer/step.rb:135:in `execute'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:464:in `block (2 levels) in map_to_context!'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:504:in `handle_mapping_errors'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:463:in `block in map_to_context!'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:457:in `each'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:457:in `map_to_context!'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/indexer.rb:582:in `block (3 levels) in process'
	from /usr/local/bundle/gems/traject-3.2.0/lib/traject/thread_pool.rb:123:in `block in maybe_in_thread_pool'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:348:in `run_task'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:337:in `block (3 levels) in create_worker'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `loop'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `block (2 levels) in create_worker'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `catch'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.5/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `block in create_worker'
```

--- a/lib/macros/date_parsing.rb
+++ b/lib/macros/date_parsing.rb
@@ -98,8 +98,8 @@ module Macros
 
     # ---------- Collection or Metadata format specific macros below, alphabetical except for additional helper methods
 
-    AUC_REGEX = Regexp.new('\d{4};') # captures the `YYYY; YYYY; YYYY; YYYY;` pattern
     AUC_DELIM = ';'
+    AUC_REGEX = Regexp.new("\\d{4}#{AUC_DELIM}") # captures the `YYYY; YYYY; YYYY; YYYY;` pattern
 
     # extracts dates from American University of Cairo data
     def auc_date_range
@@ -107,7 +107,7 @@ module Macros
         range_years = []
         accumulator.each do |val|
           range_years << val.scan(AUC_REGEX).map { |year| year.sub(AUC_DELIM, '').to_i }
-          next unless range_years.flatten!.uniq!.nil?
+          next unless range_years.flatten!&.uniq!.nil?
 
           range_years << ParseDate.range_array(ParseDate.earliest_year(val), ParseDate.latest_year(val))
         end

--- a/lib/macros/date_parsing.rb
+++ b/lib/macros/date_parsing.rb
@@ -99,7 +99,7 @@ module Macros
     # ---------- Collection or Metadata format specific macros below, alphabetical except for additional helper methods
 
     AUC_DELIM = ';'
-    AUC_REGEX = Regexp.new("\\d{4}#{AUC_DELIM}") # captures the `YYYY; YYYY; YYYY; YYYY;` pattern
+    AUC_REGEX = Regexp.new("\\d{4}#{AUC_DELIM}?") # captures the `YYYY; YYYY; YYYY; YYYY;` pattern
 
     # extracts dates from American University of Cairo data
     def auc_date_range
@@ -112,7 +112,7 @@ module Macros
           range_years << ParseDate.range_array(ParseDate.earliest_year(val), ParseDate.latest_year(val))
         end
 
-        range_years.flatten!.uniq! if range_years.any?
+        range_years.flatten!&.uniq!&.sort! if range_years.any?
         accumulator.replace(range_years)
       end
     end

--- a/spec/lib/traject/macros/date_parsing_spec.rb
+++ b/spec/lib/traject/macros/date_parsing_spec.rb
@@ -180,6 +180,7 @@ RSpec.describe Macros::DateParsing do
         expect(indexer.map_record(value: '1990-1995')).to include 'range' => [1990, 1991, 1992, 1993, 1994, 1995]
         expect(indexer.map_record(value: '990-995')).to include 'range' => [990, 991, 992, 993, 994, 995]
         expect(indexer.map_record(value: '1990; 1991; 1992;')).to include 'range' => [1990, 1991, 1992]
+        expect(indexer.map_record(value: '1990; 1991; 1992')).to include 'range' => [1990, 1991, 1992]
       end
     end
   end

